File: lily/layer-2/lily-model/src/main/java/net/bodz/lily/schema/SchemaDef.java
package net.bodz.lily.schema;

import java.util.List;

import javax.persistence.Table;

/**
 * @label Schema
 * @label.zh.cn 方案
 */
@Table(name = "schema")
public class SchemaDef
        extends AbstractDefinition<SchemaDef> {

    private static final long serialVersionUID = 1L;

    List<CategoryDef> categories;
    List<PhaseDef> phases;
    List<PriorityDef> priorities;
    List<TagGroupDef> tagSets;
    List<ParameterDef> parameters;

    public List<CategoryDef> getCategories() {
        return categories;
    }

    public void setCategories(List<CategoryDef> categories) {
        this.categories = categories;
    }

    public List<PhaseDef> getPhases() {
        return phases;
    }

    public void setPhases(List<PhaseDef> phases) {
        this.phases = phases;
    }

    public List<PriorityDef> getPriorities() {
        return priorities;
    }

    public void setPriorities(List<PriorityDef> priorities) {
        this.priorities = priorities;
    }

    public List<TagGroupDef> getTagGroups() {
        return tagSets;
    }

    public void setTagGroups(List<TagGroupDef> tagSets) {
        this.tagSets = tagSets;
    }

    public List<ParameterDef> getParameters() {
        return parameters;
    }

    public void setParameters(List<ParameterDef> parameters) {
        this.parameters = parameters;
    }

}

File: lily/@lily/packages/basic/src/net/bodz/lily/schema/meta/SchemaDef.ts
import type { List } from "skel01-core/src/lang/basetype";

import CategoryDef from "./CategoryDef";
import ParameterDef from "./ParameterDef";
import PhaseDef from "./PhaseDef";
import PriorityDef from "./PriorityDef";
import SchemaDefTypeInfo from "./SchemaDefTypeInfo";
import TagGroupDef from "./TagGroupDef";
import _SchemaDef_stuff from "./_SchemaDef_stuff";

export class SchemaDef extends _SchemaDef_stuff<SchemaDef> {

    static _typeInfo: SchemaDefTypeInfo;
    static get TYPE() {
        if (this._typeInfo == null)
            this._typeInfo = SchemaDefTypeInfo.INSTANCE;
        return this._typeInfo;
    }

    categories?: List<CategoryDef>
    parameters?: List<ParameterDef>
    phases?: List<PhaseDef>
    priorities?: List<PriorityDef>
    tagGroups?: List<TagGroupDef>

    constructor(o?: any) {
        super(o);
    }
}

export default SchemaDef;
